src/main/java/com/code_intelligence/jazzer/mutation/mutator/proto/BuilderMutatorFactory.java

```diff
@@ -243,22 +230,81 @@ public Builder detach(Builder builder) {
    * type with a message field B and B in turn has a message field of type A, then the mutators for
    * A and B will reference each other, forming a cycle.
    */
-  private final HashMap<Class<? extends Builder>, SerializingMutator<Builder>> internedMutators =
+  private final HashMap<CacheKey, SerializingMutator<? extends Builder>> internedMutators =
       new HashMap<>();
 
+  private SerializingMutator<Any.Builder> createMutatorForAny(
+      AnySource anySource, MutatorFactory factory) {
+    HashMap<String, Integer> typeUrlToIndex = new HashMap<>(anySource.value().length);
+    for (int i = 0; i < anySource.value().length; i++) {
+      Message defaultInstance = TypeLibrary.getDefaultInstance(anySource.value()[i]);
+      typeUrlToIndex.put(getTypeUrl(defaultInstance), i);
+    }
+
+    return assemble(mutator
+        -> internedMutators.put(new CacheKey(Any.Builder.class, anySource), mutator),
+        Any.getDefaultInstance()::toBuilder, makeBuilderSerializer(Any.getDefaultInstance()),
+        ()
+            -> mutateSumInPlace(
+                // Corpus entries may contain Anys with arbitrary (and even invalid) messages, so we
+                // fall back to mutating the first message type if the type isn't recognized.
+                (Any.Builder builder)
+                    -> typeUrlToIndex.getOrDefault(builder.getTypeUrl(), 0),
+                stream(anySource.value())
+                    .map(messageClass -> {
+                      SerializingMutator<Message> messageMutator =
+                          (SerializingMutator<Message>) factory.createOrThrow(notNull(
+                              withExtraAnnotations(asAnnotatedType(messageClass), anySource)));
+                      return mutateProperty(
+                          (Any.Builder anyBuilder)
+                              -> {
+                            try {
+                              return anyBuilder.build().unpack(messageClass);
+                            } catch (InvalidProtocolBufferException e) {
+                              // This can only happen if the corpus contains an invalid Any.
+                              return TypeLibrary.getDefaultInstance(messageClass);
+                            }
+                          },
+                          messageMutator,
+                          (Any.Builder any, Message message) -> {
+                            any.setTypeUrl(getTypeUrl(message));
+                            any.setValue(message.toByteString());
+                          });
+                    })
+                    .toArray(InPlaceMutator[] ::new)));
+  }
+
+  private static String getTypeUrl(Message message) {
+    return "type.googleapis.com/" + message.getDescriptorForType().getFullName();
```

Comment:
Could you add a comment how the message type url relates to the builder type url? Or perhaps a link to https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/any.proto#L94

Reply:
Done.